hazelcast/src/test/java/com/hazelcast/client/impl/MemberImplTest.java

```diff
@@ -0,0 +1,196 @@
+        assertEquals(Long.MAX_VALUE, longValue.longValue());
+
+        Float floatValue = member.getFloatAttribute("floatKey");
+        assertNotNull(floatValue);
+        assertEquals(Float.MAX_VALUE, floatValue, 0.000001);
+
+        Double doubleValue = member.getDoubleAttribute("doubleKey");
+        assertNotNull(doubleValue);
+        assertEquals(Double.MAX_VALUE, doubleValue, 0.000001);
+    }
+
+    @Test
+    public void testConstructor_withMemberImpl() {
+        member = new MemberImpl(new MemberImpl(address));
+
+        assertMemberImpl();
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testStringAttribute() {
+        member = new MemberImpl(address);
+        member.setStringAttribute("stringKey", "stringValue");
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testBooleanAttribute() {
+        member = new MemberImpl(address);
+        member.setBooleanAttribute("booleanKeyTrue", true);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testByteAttribute() {
+        member = new MemberImpl(address);
+        member.setByteAttribute("byteKey", Byte.MAX_VALUE);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testShortAttribute() {
+        member = new MemberImpl(address);
+        member.setShortAttribute("shortKey", Short.MAX_VALUE);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testIntAttribute() {
+        member = new MemberImpl(address);
+        member.setIntAttribute("intKey", Integer.MAX_VALUE);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testLongAttribute() {
+        member = new MemberImpl(address);
+        member.setLongAttribute("longKey", Long.MAX_VALUE);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testFloatAttribute() {
+        member = new MemberImpl(address);
+        member.setFloatAttribute("floatKey", Float.MAX_VALUE);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testDoubleAttribute() {
+        member = new MemberImpl(address);
+        member.setDoubleAttribute("doubleKey", Double.MAX_VALUE);
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testRemoveAttribute() {
+        member = new MemberImpl(address);
+        member.removeAttribute("removeKey");
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testRemoveAttribute_onRemoteMember() {
+        member = new MemberImpl(address);
+        member.removeAttribute("remoteMemberRemove");
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void testSetAttribute_onRemoteMember() {
+        member = new MemberImpl(address);
+        member.setStringAttribute("remoteMemberSet", "wontWork");
+    }
+
+    private void assertMemberImpl() {
```

Comment:
What's your suggestion?

(your comment got outdated, since I just removed two useless test methods which were copied from the other `MemberImplTest` class).